azurelinuxagent/common/protocol/extensions_goal_state.py

```diff
@@ -16,14 +16,432 @@
 #
 # Requires Python 2.6+ and Openssl 1.0+
 
-import re
+import json
+
+from collections import defaultdict
+
+import azurelinuxagent.common.logger as logger
+from azurelinuxagent.common.event import add_event, WALAEventOperation
+from azurelinuxagent.common.exception import ExtensionConfigError
+from azurelinuxagent.common.future import ustr
+from azurelinuxagent.common.protocol.restapi import Extension, ExtHandler, ExtHandlerList, ExtHandlerVersionUri, VMAgentManifest, \
+    VMAgentManifestList, VMAgentManifestUri, InVMGoalStateMetaData, RequiredFeature, ExtensionState
+from azurelinuxagent.common.utils.textutil import parse_doc, findall, find, findtext, getattrib, gettext
 
 
 class ExtensionsGoalState(object):
-    def __init__(self, etag, vm_settings):
-        self.etag = etag
-        self.vm_settings = vm_settings
+    def __init__(self):
```

Comment:
most of this code comes straight from the previous ExtensionsConfig, with minor changes in __init__() and from_extensions_config()